tools/nixpkgs-hammer

```diff
@@ -44,6 +53,22 @@ def ast_checks(attr_positions: Dict[str, str], excluded_rules: List[str]) -> Dic
     return dict(result)
 
 
+def log_checks(attr_output_paths: Dict[str, str], excluded_rules: List[str]) -> Dict[str, List]:
+    rules = get_log_check_programs() - set(excluded_rules)
+    if len(rules) == 0:
+        return {}
+
+    report = defaultdict(list)
+    for attr_name, out_path in attr_output_paths.items():
+        p1 = subprocess.run(["nix", "--experimental-features", "nix-command", "log", out_path], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
We need to ensure the path is built first before we can query it for logs.

Reply:
Can't we just check the return code?

Reply:
`nix log` returns 1 if you pass it a store path or attribute path that has not been built and is not available in binary cache.